codec/src/main/java/io/netty/handler/codec/compression/Lz4FrameEncoder.java

```diff
@@ -254,6 +259,46 @@ protected void encode(ChannelHandlerContext ctx, ByteBuf in, ByteBuf out) throws
         }
     }
 
+    @Override
+    protected ChannelPromise encode(ChannelHandlerContext ctx, ByteBuf msg, ByteBuf out, ChannelPromise promise)
+            throws Exception {
+        try {
+            encode(ctx, msg, out);
+        } catch (Exception e) {
+            if (bufferPromise != null) {
+                bufferPromise.tryFailure(e);
+            }
+            throw e;
+        }
+
+        if (bufferPromise == null) {
+            if (finished) {
+                return promise;
+            }
+            if (buffer.isReadable()) {
+                bufferPromise = promise;
+                return ctx.newPromise();
```

Comment:
This doesn't look right... if the returned promise will be failed we will never fail `promise` or `bufferPromise` which I think we should